indexer/src/main.rs

```diff
@@ -121,24 +129,68 @@ async fn main() -> anyhow::Result<()> {
         )
     })?;
 
-    let mut sink: CardanoSink = match config.sink {
-        SinkConfig::Cardano { .. } => CardanoSink::new(config.sink, exec_plan).await?,
-        _ => todo!("not supported yet"),
+    let (network, mut sink) = match config.sink {
+        SinkConfig::Cardano { ref network, .. } => (
+            network.clone(),
+            CardanoSink::new(config.sink, exec_plan).await?,
+        ),
     };
 
     let start_from = sink.start_from(config.start_block).await?;
 
-    let mut engine = match &config.source {
+    match &config.source {
         SourceConfig::Oura { .. } => {
             let source = OuraSource::new(config.source, start_from.clone())?;
-            engine::FetchEngine::new(source, sink)
+            let start_from = start_from.into_iter().next().unwrap();
+
+            main_loop(source, sink, start_from).await
         }
-        _ => todo!("not supported yet"),
+        SourceConfig::CardanoNet { relay } => {
```

Comment:
for cardano-net we can put the network here as well. I know we're duplicating the parameters, but better to duplicate than to break the abstraction

Reply:
I'm not sure I see the abstraction. I think the network should be a global config. There is no reason you would want one network for the sink and another for the source.

Reply:
potentially we can support other blockchains, and `network` might not exist as a parameter there